transformer_engine/common/CMakeLists.txt

```
@@ -49,19 +65,20 @@ if(NOT EXISTS "${CUDNN_FRONTEND_INCLUDE_DIR}")
             "Try running 'git submodule update --init --recursive' "
             "within the Transformer Engine source.")
 endif()
-
-# Configure dependencies
-target_link_libraries(transformer_engine PUBLIC
-                      CUDA::cublas
-                      CUDA::cuda_driver
-                      CUDA::cudart
-                      CUDA::nvrtc
-                      CUDA::nvToolsExt
-                      CUDNN::cudnn)
-target_include_directories(transformer_engine PRIVATE
-                           ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
 target_include_directories(transformer_engine PRIVATE "${CUDNN_FRONTEND_INCLUDE_DIR}")
 
+# Check for MPI (optional)
+option(UB_MPI_BOOTSTRAP "Bootstrap userbuffers with MPI." OFF)
+message(STATUS "Userbuffers MPI bootstrap: ${UB_MPI_BOOTSTRAP}")
+if (UB_MPI_BOOTSTRAP)
+     find_package(MPI REQUIRED)
```

Change from https://github.com/NVIDIA/TransformerEngine/pull/828:
```suggestion
     find_package(MPI REQUIRED COMPONENTS CXX)
```